docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+..
+   Author: Adam Turner
+
+   We can't use :pep:`X` references in this document, as they are relative from
```

Comment:
```suggestion
   We can't use :pep:`NNN` references in this document, as they are relative to
```

Did you mean this instead? Also, use a clearer placeholder.

Reply:
The phrasing is poor. What I was trying to express is that using `:pep:` directives *within the PEPs repo* is special, as to make local builds etc work, we define the PEP url scheme to be relative rather than absolute. These relative links are *from* the content root *to* a PEP, and using one in the `docs` subdirectory doesn't work.

It is a phrasing issue that will entirely go away if hosted on the dev guide!

Hopefully that makes a bit more sense -- I'll make the X -> N change now, but leave the other one to see if there's a way of more cleanly expressing the concept above.

A

Reply:
> My phrasing was poor. What I was trying to express is that using :pep: directives within the PEPs repo is special, as to make local builds etc work, we define the PEP url scheme to be relative rather than absolute. These relative links are from the content root to a PEP, and using one in the docs subdirectory doesn't work.

Yeah, with some effort, I figured out what you probably meant due to being familiar with the type of issue in question, having run into similar problems myself, but it was pretty unclear and confusing just from the text here, and in general "relative from" isn't really used in this context. Instead, what about:

```
We can't use :pep:`NNN` references in this document, as they use links
relative to the current document, which doesn't work in a subdirectory
like this one.
```

Also, wouldn't a better solution be to make the ``:pep:`NNN``` role assume paths relative to the project root (i.e. the `conf.py` directory of the build) rather than the current document? Or is that not really easily possible (I seem to recall running into troubles with that same issue)

> It is a phrasing issue that will entirely go away if hosted on the dev guide!

To be fair, so will internal links to the PEPs in general :slightly_smiling_face:

Reply:
> a better solution be to make the :pep: role assume paths relative to the project root

Not possible (as far as I can tell) due to Sphinx and HTML(/reverse proxies/the URI protocol?). Sphinx (Docutils, really) treats the pattern literally as a string that it interpolates with the PEP number, doing no path-like operations to it. We could make it an absolute link with no domain, but that then breaks github pages &c, as they host the PEPs at a subdir as a reverse proxy, not the content root. So for the greatest portability we have to sacrifice using `:pep:` roles in subdirs :( 

A